packages/eslint-plugin/src/rules/prefer-readonly.ts

```diff
@@ -173,6 +173,38 @@ export default createRule<Options, MessageIds>({
       };
     }
 
+    function getTypeAnnotationForViolatingNode(
+      node: TSESTree.Node,
+      type: ts.Type,
+      initializerType: ts.Type,
+    ) {
+      const annotation = checker.typeToString(type);
+
+      // verify the about-to-be-added type annotation is in-scope
+      if (tsutils.isTypeFlagSet(initializerType, ts.TypeFlags.EnumLiteral)) {
```

Comment:
I wrote about this on https://github.com/typescript-eslint/typescript-eslint/issues/1056#issuecomment-2562979820, but on some occasions (as far as I know only enums), the rule won't be able to add a type annotation:

```ts
// foo.ts
enum Foo {
  Bar,
  Bazz,
}

export const bar = Foo.Bar;

// index.ts
import { bar } from './foo';

export class Test {
  private foo = bar;
}
```

I chose not to add a type annotation on these cases, but an alternative I think is also valid is to have this be a suggestion instead of an auto-fix.

Would be happy to to hear opinions on this.